src/main/java/com/google/sps/servlets/AddOrganizationServlet.java

```diff
@@ -37,21 +56,29 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String orgUrl = request.getParameter("url-link");
     String orgDescription = request.getParameter("description");
 
-    //TODO: get timestamp with transactions instead
-    long timestampMillis = System.currentTimeMillis();
+    /* MillisecondSinceEpoch represent the number of milliseconds that have passed since
+     * 00:00:00 UTC on January 1, 1970. It ensures that all users are entering a representation
+     * of time that is independent of their time zone */
+    long millisecondSinceEpoch = Instant.now().toEpochMilli();
 
     // when suppliers are added, Entity kind will be from a parameter- for now is hardcoded
     Entity newOrganization = new Entity("Distributor");
 
-    ArrayList<String> moderatorList = new ArrayList<String>();
-    moderatorList.add("anon creator"); //dummy value for before auth gets set up
+    /* This implementation stores history entries as embedded entities instead of custom objects
+     * because it is much simpler that way */
+    ArrayList changeHistory = new ArrayList<>();
+    
+    HistoryManager history = new HistoryManager();
+
+    changeHistory.add(history.recordHistory(userId, "Organization was registered", millisecondSinceEpoch));
+    newOrganization.setProperty("changeHistory", changeHistory);
 
-    // TODO change this to our History object- for prototyping just using strings
-    ArrayList<String> changeHistory = new ArrayList<String>();
-    changeHistory.add("Organization was registered");
 
-    newOrganization.setProperty("creationTimeStamp", timestampMillis);
-    newOrganization.setProperty("lastEditTimeStamp", timestampMillis);
+    ArrayList<String> moderatorList = new ArrayList<String>();
+    moderatorList.add(username);
+
+    newOrganization.setProperty("creationTimeStampMillis", millisecondSinceEpoch);
```

Comment:
Leave a TODO to replace 80-84 with sarah's changes to update the organization from a request.